core/templates/services/contextual/url.service.ts

```diff
@@ -149,15 +149,15 @@ export class UrlService {
     return null;
   }
 
-  getSubtopicUrlFragmentFromLearnerUrl(): string {
+  getSubtopicUrlFragmentFromLearnerUrl(): string | undefined {
     let pathname = this.getPathname();
     if (pathname.startsWith('/learn') && pathname.includes('/revision')) {
       return decodeURIComponent(pathname.split('/')[5]);
     }
     throw new Error('Invalid URL for subtopic');
   }
 
-  getClassroomUrlFragmentFromLearnerUrl(): string {
+  getClassroomUrlFragmentFromLearnerUrl(): string | undefined {
```

Comment:
why the `get` method is returning `undefined`?

Reply:
Same case as above.

Reply:
Shouldn't it be void instead of undefined?

Reply:
Both void and undefined work in this case. But maybe using void makes it more readable. Thanks!

Reply:
I used _undefined_ because the return of void functions could be considered _undefined_. I had not tried _void_. But now changing it and doing the tests, I got this error in more than 10 files:

![image](https://user-images.githubusercontent.com/46513107/114818035-ab491400-9d80-11eb-9302-e7fbb50e8238.png)

Should I continue to use _undefined_, or should I change it to _void_ and fix the error I showed in all the files where this occurs?

Reply:
PTAL @srijanreddy98

Reply:
Here I try a function that doesn't have a "return", and the type I get is _undefined_

![image](https://user-images.githubusercontent.com/46513107/114819251-cd439600-9d82-11eb-9e62-72d1b118cd3e.png)

Reply:
Thanks! Approved.